utility/doc/CHANGELOG.md

```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.0.0.14] - 2022-12-20
```

Comment:
We don't actually have a CHANGELOG  for infra/tooling.

Putting this under `build` seems wrong (irrelevant), but creating a CHANGELOG under `root` also seems wrong (it's just tooling).

Since I don't have a good answer for it for now. Seems that your script checks for a CHANGELOG under root, which we don't have yet. DO you want to add it?

Reply:
This was a question I posed in the original PR, I don't see the need to create a CHANGELOG simply for this one piece of tooling. I have removed the root directory check as there is nothing there. I think it is probably better _in my opinion_ to remove this check as there isn't really a need for it.

If you want me to remove the changes I made to `utility/doc/CHANGELOG.md` I can but I don't see the point in making a new root level `CHANGELOG.md` as it probably will not see any updates besides this - there is no top-level code. All the other modules have their own as well so let me know what you think about this. I can if that's what you are looking for.

Reply:
Adding the root changelog would mean that for any update to a module that requires `go.mod` or `go.sum` to change then both the module and root `CHANGELOG.md` files would need to be included - which would be an pointless repetition. We could get around this by having the script ignore all files in the root directory - as they don't require a changelog really. But this is the same as not checking it at all...

Reply:
> I don't see the need to create a CHANGELOG simply for this one piece of tooling

Agreed

> . I think it is probably better in my opinion to remove this check as there isn't really a need for it.

Since I'm not 100% sure at the moment, but I can see us adding a root changelog soon (for main releases), can you  add a `ROOT_CHANGELOG_ENABLE=0 ` with a comment explaining that it should be enabled once a root CHANGELOG is added. We get to keep the code and document why it is off at the moment.

Reply:
> Scan you add a `ROOT_CHANGELOG_ENABLE=0 ` with a comment explaining that it should be enabled once a root CHANGELOG is added. We get to keep the code and document why it is off at the moment.

So instead of this I have the script check the root level _ONLY_ for a `CHANGELOG.md` file. This avoids having to do the full root level check/ignore certain files but allows for the root-level `CHANGELOG.md` - _only when included_ to be validated.

This does mean there can never be an error situation for not including it in a commit. However it means that there doesn't need to be any code changes when the CHANGELOG gets implemented.

I would be happy to implement a checking system when the CHANGELOG is made and the circumstances defining when and when not to include it in a commit are clearly defined. But for now if its included it will be validated (date+version) like the rest but if not there will be no errors.